asset/src/kafka_sender/processor.ts

```diff
@@ -6,45 +6,110 @@ import {
     ConnectionConfig,
     getValidDate,
     isString,
+    has,
 } from '@terascope/job-components';
 import * as kafka from 'node-rdkafka';
 import { KafkaSenderConfig } from './interfaces';
 import { ProducerClient, ProduceMessage } from '../_kafka_clients';
 
-export default class KafkaSender extends BatchProcessor<KafkaSenderConfig> {
+interface Endpoint {
     producer: ProducerClient;
+    data: any[];
+}
+
+interface TopicMap {
+    [key: string]: Endpoint;
+}
+
+export default class KafkaSender extends BatchProcessor<KafkaSenderConfig> {
     private _bufferSize: number;
+    topicMap: TopicMap = {};
 
     constructor(
         context: WorkerContext,
         opConfig: KafkaSenderConfig,
         executionConfig: ExecutionConfig
     ) {
         super(context, opConfig, executionConfig);
+        const { topic, size, connection_map: connectionMap } = opConfig;
 
         const logger = this.logger.child({ module: 'kafka-producer' });
 
-        this._bufferSize = this.opConfig.size * 5;
+        this._bufferSize = size * 5;
+
+        for (const keyset of Object.keys(connectionMap)) {
+            const client = this.createClient(connectionMap[keyset]);
+            const keys = keyset.split(',');
 
-        this.producer = new ProducerClient(this.createClient(), {
-            logger,
-            topic: this.opConfig.topic,
-            bufferSize: this._bufferSize,
-        });
+            for (const key of keys) {
+                const newTopic = key === '*' ? topic : `${topic}-${key}`;
+                const producer = new ProducerClient(client, {
+                    logger,
+                    topic: newTopic,
+                    bufferSize: this._bufferSize,
+                });
+
+                this.topicMap[key] = {
+                    producer,
+                    data: []
+                };
+            }
+        }
     }
 
     async initialize() {
         await super.initialize();
-        await this.producer.connect();
+        const initList = [];
+
+        for (const [, { producer }] of Object.entries(this.topicMap)) {
+            initList.push(producer.connect());
+        }
+
+        await Promise.all(initList);
+    }
+
+    private _cleanupTopicMap() {
+        for (const [, config] of Object.entries(this.topicMap)) {
+            config.data = [];
+        }
     }
 
     async shutdown() {
-        await this.producer.disconnect();
+        const shutdownList = [];
+
+        for (const [, { producer }] of Object.entries(this.topicMap)) {
+            shutdownList.push(producer.disconnect());
+        }
+
+        await Promise.all(shutdownList);
         await super.shutdown();
     }
 
     async onBatch(batch: DataEntity[]) {
-        await this.producer.produce(batch, this.mapFn());
+        const senders = [];
+
+        for (const record of batch) {
+            const partition = record.getMetadata('_partition');
+
+            if (has(this.topicMap, partition)) {
```

Comment:
With a map this would be more readable:

```suggestion
            if (this.topicMap.has(partition)) {
```